### PR TITLE
Add comments

### PR DESCRIPTION
**Pull request by justxuewei**
**What this PR does**:

- add comments
- use uniform receiver name for `Http2Server`
- use `data, ok := <- chan` to determined if the channel is closed
- move `example/http2/client/client.go` to `example/http2/client/cmd/client.go`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/http2/server.go [hunk 1]
```diff
@@ -25,7 +25,12 @@ import (
 	tConfig "github.com/dubbogo/triple/pkg/http2/config"
 )
 
-type Http2Handler func(path string, header http.Header, recvChan chan *bytes.Buffer, sendChan chan *bytes.Buffer, ctrlch chan http.Header, errCh chan interface{})
+// Http2Handler uses to relay data to upper layer and receive data from upper layer as well.
```
Comment (AlexStocks) [inline pkg/http2/server.go]: Http2Handler relays data to upper layer and receives data from upper layer as well.

File: pkg/http2/server.go [hunk 2]
```diff
@@ -25,7 +25,12 @@ import (
...
+// recvChan relays body data to upper layer, see also readChan defined in http2HandleFunction()
```
Comment (AlexStocks) [inline pkg/http2/server.go]: see also readChan defined in http2HandleFunction()

我读不懂啥意思

Comment (xavier-niu) [reply pkg/http2/server.go]: 我改了下

File: pkg/http2/server.go [hunk 3]
```diff
@@ -25,7 +25,12 @@ import (
 	tConfig "github.com/dubbogo/triple/pkg/http2/config"
 )
 
-type Http2Handler func(path string, header http.Header, recvChan chan *bytes.Buffer, sendChan chan *bytes.Buffer, ctrlch chan http.Header, errCh chan interface{})
+// Http2Handler relays data to upper layer and receives data from upper layer as well.
+// recvChan relays data from lower layer to upper layer, generally lower layer refers to http body data
```
Comment (xavier-niu) [inline pkg/http2/server.go:29]: @AlexStocks 我改了下

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**